### PR TITLE
ISPN-11323 solate JGroupsTransport MBean references to private methods

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-11323

### CODE REVIEW

**Comment by karesti** [comment]
@wburns LGTM, waiting CI to finish

**Comment by anistor** [comment]
I assume this is to be backported to 10.1.x also, @wburns ?

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java [hunk 1]
```diff
@@ -523,27 +523,31 @@ protected void startJGroupsChannelIfNeeded() {
...
+      startJmxIfNeeded(clusterName);
       if (!connectChannel) {
          // the channel was already started externally, we need to initialize our member list
          receiveClusterView(channel.getView());
       }
       CLUSTER.localAndPhysicalAddress(clusterName, getAddress(), getPhysicalAddresses());
    }
 
+   // This needs to stay as a separate method to allow for substitution for Substrate
+   private void startJmxIfNeeded(String clusterName) {
```
Comment (anistor) [inline core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java]: No, this does not start JMX :) Please name it ```registerMBeans```.

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java [hunk 2]
```diff
@@ -807,6 +809,13 @@ public void stop() {
       }
    }
 
+   // This needs to stay as a separate method to allow for substitution for Substrate
+   private void stopJmxIfNeeded(String clusterName) throws Exception {
```
Comment (anistor) [inline core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java]: ```unregisterMBeans```

**Comment by anistor** [comment]
> @wburns LGTM, waiting CI to finish

no, it's not :)

**Comment by wburns** [comment]
> > @wburns LGTM, waiting CI to finish
> 
> no, it's not :)



> I assume this is to be backported to 10.1.x also, @wburns ?

Ofc it will be.

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java (hunk 1 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java]: Okay.

File: core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java (hunk 2 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java]: Okay.

**Comment by wburns** [comment]
> > @wburns LGTM, waiting CI to finish
> 
> no, it's not :)

Not really helpful :)

**Comment by wburns** [comment]
Updated

**Comment by anistor** [comment]
> > I assume this is to be backported to 10.1.x also, @wburns ?
> 
> Ofc it will be.

Please do not forget to backport this to 10.1.x

**Comment by wburns** [comment]
Was next on my list - was finishing a different backport first :)

**Comment by anistor** [comment]
I see the jira is still open. I'll close it.

**Pull request merged**